tools/CAT/src/pages/EditPage/index.tsx

```diff
@@ -1,115 +1,84 @@
+import { useParams } from 'react-router-dom';
+import {
+	deletePage,
+	updatePage,
+	fetchPageByPath,
+} from '@/app/features/pages/pageThunks';
+import { useAppDispatch, useAppSelector } from '@/app/hooks';
+import {
+	getPageDataSelector,
+	getPageLoadingSelector,
+	getPageErrorSelector,
+} from '@/app/features/pages/pageSlice';
+import { useNavigate } from 'react-router-dom';
+import { PageDto } from '@/models/api';
+import { Formik, ErrorMessage, Form } from 'formik';
+import { updatePageSchema } from '@/validation/update.page.schema';
+import {
+	getСategoryDataSelector,
+	getСategoryLoadingSelector,
+} from '@/app/features/categories/categorySlice';
+import { fetchCategoriesData } from '@/app/features/categories/categoryThunks';
+import { showErrorNotif } from '@/providers/notify';
 
-interface EditPageProps {
-	handleClose: () => void;
-	parentUrl: string;
-	id: string;
-	subCategoryId: string;
-}
+export default function EditPage() {
+	const { category, subcategory, page } = useParams();
+	const viewUrl = `${
+		import.meta.env.VITE_API_URL
+	}/${category}/${subcategory}`;
 
-export default function EditPage({
-	handleClose,
-	parentUrl,
-	id,
-	subCategoryId,
-}: EditPageProps) {
-	const { pageData, setPageData } = useInitialPageState({
-		id,
-		subCategoryId,
-	});
-	const [deletePage] = useDeletePageMutation();
-	const { data, isLoading } = useGetPageQuery(id);
+	const dispatch = useAppDispatch();
+	const pageData = useAppSelector(getPageDataSelector);
+	const isLoading = useAppSelector(getPageLoadingSelector);
+	const errorPage = useAppSelector(getPageErrorSelector);
+	const categoriesData = useAppSelector(getСategoryDataSelector);
+	const categoriesLoading = useAppSelector(getСategoryLoadingSelector);
 
-	const [updatePage] = useUpdatePageMutation();
+	const initialPage: PageDto = pageData;
 
-	const handleChangeTitle = ({ target }) => {
-		const { value } = target;
-		setPageData({ ...pageData, title: value });
-	};
+	const navigate = useNavigate();
 
 	useEffect(() => {
-		setPageData({
-			...pageData,
-			id: (data as { id: string })?.id ?? pageData.id,
-			title: (data as { title: string })?.title ?? pageData.title,
-			url: (data as { url: string })?.url ?? pageData.url,
-			content: (data as { content: string })?.content ?? pageData.content,
-			template: {
-				...pageData.template,
-				label:
-					(data as { template?: { label?: string } })?.template
-						?.label ?? pageData.template.label,
-				type:
-					(data as { template?: { type?: number } })?.template
-						?.type ?? pageData.template.type,
-				image: {
-					...pageData.template.image,
-					url:
-						(data as { template?: { image?: { url?: string } } })
-							?.template?.image?.url ??
-						pageData.template.image.url,
-				},
-			},
-		});
-	}, [data]);
-
-	const handleChangeUrlPage = ({ target }) => {
-		const { value } = target;
-		setPageData({ ...pageData, url: value });
-	};
+		categoriesData.length === 0 && dispatch(fetchCategoriesData());
```

Comment:
better to use if statement here, absolute AND we use in cases when we wanna return smth, there we just calling the dispatch